pdns/dnsdist-lua2.cc

```diff
@@ -1291,4 +1295,27 @@ void moreLua(bool client)
         }
 #endif /* HAVE_NET_SNMP */
       });
+
+    g_lua.writeFunction("setPoolServerPolicy", [](ServerPolicy policy, string pool) {
+        setLuaSideEffect();
+        auto localPools = g_pools.getCopy();
+        setPoolPolicy(localPools, pool, std::make_shared<ServerPolicy>(policy));
+      });
```

Comment:
The issue here is that you are updating a local copy of the pools but not updating the global state. It won't work if the pool doesn't already exists and `setPoolPolicy()` creates it, because it will only exist in the local copy. You need to do `g_pools.setState(localPools)` afterwards. Same issue in `setPoolServerPolicyLua()`.